### DISCUSSION TITLE
Ezra Bible App on Linux Mint

### DISCUSSION THREAD

**Discussion by Opv12** [ezra-bible-app/ezra-bible-app#952] [General]
Just installed the app on Linux Mint.  I like the ability to tag verses much more than the traditional verse lists.  However, I'm having difficulty with certain aspects of the functionality and would like to know if it's the app or my operating system.  Before I invest a lot of time creating a gazillion tags I wanted to learn a bit more about the app.  I'm particularly interested in the ability to select verses and to copy those verses to an external editor, while retaining all of the relevant formatting of the verse, i.e., Christ's wods in red and the italicized words, as applicable.  The problems I'm experiencing include:
1. Inability to "select all" verses when a Tag List is displayed.  I seem to have to either manually select each verse individually by holding the CTRL button down or drag and drop ONLY the currenntly visible verses.  The view does not scroll when I attempt to grab all verses included in the list.  Is there a way to select all?  CTRL-A does not seem to work.
2. All of my efforts to copy verses and paste them into an external editor result in my losing all formatting.  I've tried copy and paste from right click, as well as CTRL-C and CTRL-V.  The result is the same.  
Are the issues I've described by design or am I overlooking something?  If by design, are there plans for the described functionality to be added, and if so is there a projected timeline as to when said functioanlity is expected?

**Comment by tobias-klein** [comment]
Hi @Opv12,

Thanks for your interest in Ezra Bible App and thanks for your feedback!

1. A possibility to select all verses of a tagged verse list is not implemented. So far it did not occur to me that one may want to assign the same tag to a whole big list of verses from a tagged verse list. Such a function is a available for search results (there is a button *Select all search results*) and it could be also added to tagged verse lists.
2. Copying verses to the clipboard currently only copies plain text, no formatting. What we could do is add an option that makes the behavior configurable. If the option is enabled we could instead copy html (that's the markup used for the formatting). Then it depends on the tool where paste the text whether the formatting is correctly shown.

The next release is the upcoming weekend. It is too late for that. The next release after that would be probably beginning of next year. I could check whether those two enhancements could be added then.

**Reply by Opv12** [reply]
Thanks.  Yes, I tend to tag not only individual verses but all of the verses pertaining directly to that verse so as to provide a more complete context, as I like to export (or copy/paste) the fuller context to an editor when I writie about the topic.  That's the reason my question extended to retaining the formatting.  Similarly, it's one of the reasons I've tended to avoid taking advantage of the note-taking functionality built in to Bible software packages, as my notes tend to end up being accessible only while I'm using the application, i.e., when I've viewing a particular passage.  My objective extends more to facilitating my writing.  I haven't found a convenient way to gather everything pertaining to a particular topic (verse list (or tagged verses), linked notes, etc) from within the applications into one place so that all of the individual pieces can be organized and developed into a final product.  I really like the possibilities apparent in the Ezra Bible App.

**Reply by tobias-klein** [reply]
@Ovp12

1) This is implemented now and will be available with the next release (see #958 and #959)
2) I have raised issue #962 for this now. I'll see whether this can also be added to the next release.

**Reply by Opv12** [reply]
Thank you!

**Comment by Opv12** [comment]
I happend to think of another aspect related to tagging that perhaps one day can be given consideration.  If there is an unknown number of verses in an extend passage pertaining to a specific topic that I'd like to tag, it would great to be able to select the first verse, then scroll down and find the last related verse and SHIFT-CLICK to select all of the interim verses.

**Reply by tobias-klein** [reply]
Hi @Opv12! This particular enhancement has been implemented now (see #957). It will be included in the next release, which will be published in January 2024.

**Reply by Opv12** [reply]
Wow! That was fast.  Thanks!
